lib/game.rb

```diff
@@ -0,0 +1,81 @@
+class Game
+
+  def initialize(state=Array.new(9, ""))
+    @state = state
+  end
+
+  def set_mark(position, marker)
+    index = position - 1
+    @state[index] = marker
+  end
+
+  def get_mark_at(position)
+    index = position - 1
+    @state[index]
+  end
+
+  def set_marks(row1, row2, row3)
+    for index in 0..2 do
+      @state[index] = row1[index]
+      @state[index + 3] = row2[index]
+      @state[index + 6] = row3[index]
+    end
+  end
+
+  def print_board
+    for index in 0..8 do
+      val = (@state[index] != '') ? @state[index]: "_"
+      print "|" + val.to_s + "|"
+      if index == 2 || index == 5
+        print "\n"
+      end
+    end
+  end
```

Comment:
Can you avoid direct calls to "print" inside this class?  What sort of abstraction could you build for the I/O.  This is quite hard to test as it is. (And I noticed there are none for this method) :)